packages/studio-base/src/panels/ThreeDeeRender/ModelCache.ts

```diff
@@ -105,7 +105,7 @@ export class ModelCache {
 
     // Check if this is a STL file based on content-type or file extension
     if (STL_MIME_TYPES.includes(contentType) || /\.stl$/i.test(url)) {
-      return this.#loadSTL(url, buffer, this.options.meshUpAxis);
+      return this.#loadSTL(url, buffer.buffer.slice(buffer.byteOffset), this.options.meshUpAxis);
```

Comment:
Would like to avoid this copy here, but not sure how. It otherwise doesn't work if the Uint8Array has a byteOffset > 0

Reply:
Is that an error with the three.js STLLoader? Probably the "right" thing to do is make a PR upstream to fix it :)

Reply:
At least for now we should add a comment to explain why this slice is needed

Reply:
> Is that an error with the three.js STLLoader? Probably the "right" thing to do is make a PR upstream to fix it :)

I wouldn't say it's an error, the API just expects you to pass in an ArrayBuffer instance where the first byte is already part of the STL. This wouldn't be the case if we have a byteOffset > 0. 

```suggestion
      // Create a copy of the array buffer to respect the `byteOffset` value as the
      //  underlying three.js STLLoader only accepts an ArrayBuffer instance.
      return this.#loadSTL(url, buffer.buffer.slice(buffer.byteOffset), this.options.meshUpAxis);
```

Reply:
We probably need to respect `buffer.byteLength` then too, right?

Reply:
> We probably need to respect buffer.byteLength then too, right?

You are probably right :sweat_smile: 

```suggestion
      // Create a copy of the array buffer to respect the `byteOffset` and `byteLength` value as
      // the underlying three.js STLLoader only accepts an ArrayBuffer instance.
      return this.#loadSTL(
        url,
        buffer.buffer.slice(buffer.byteOffset, buffer.byteOffset + buffer.byteLength),
        this.options.meshUpAxis,
      );
```